### DISCUSSION TITLE
All-in-one Virtual Console for TIC-80

### DISCUSSION THREAD

**Discussion by revdeluxe** [nesbox/TIC-80#1731] [Ideas]
**What is [EmulationStation](https://emulationstation.org/)?**

-  EmulationStation is a graphical and theme able, emulator front-end that allows you to access all your favorite games in one place.

**Features**

-  Works with any controller.
-  Has Themes. to complete your setup aesthetics
-  And a ROM scraper(Although i don't think tic-80 has support for this one, but stay tuned i **might** make one.)
> Make sure your .tic games is posted on the [tic80](https://tic80.com/play) because that's the only place, that is accessible for me profile the cartridges in the rom scraper. Please include Screenshots it will serve as the rom's thumbnail.

**Why ???**
- I know that retroarch already have libretro-tic80, but hear me out. commercially retroarch aesthetics only gives you that console like interfaces that doesn't fit to your setup.
- Also i know that Emulationstation supports any software emulator, apps and virtual console, and this makes it the best option to put every game in your device in one place. for you and others to access with ease
- Why not.
- EmulationStation makes a best interface for a handheld device, plus the theme is customizable.

**Setup**
- Download [EmulationStation](https://emulationstation.org/)/[Retropie](https://retropie.org.uk) from the website. either of which.
- Install it.
- Find/Locate the es_system.cfg, Assuming that you installed tic-80 in your system, and not a libretro version.
- Find/Locate the TIC-80 home folder. Perform this command `folder`. your file manager should popup and then in the same directory make a folder named "GAMES" then copy the path.
- Edit the es_system.cfg then put this code inside the sytemlist tag.
```
<system> 
    <name>TIC-80</name>
    <fullname>Virtual Console TIC-80</fullname>
    <path>##--------PASTE THE PATH YOU'D COPIED EARLIER---------##</path>
    <extension>.tic</extension> 
    <command>tic80 --skip --cmd="cd games & load %BASENAME% & run"</command>
    <platform>TIC-80</platform>
    <theme>tic80</theme>
</system>
```
- RUN and test if everything works.

**Remarks and other Notes**
- The theme varies to everyone. so i'll make everything in a clear .xml and mostly profiled.
- there will be a follow up for this one so yeah stay-tuned

**Some Screenshots and clips**
- As you can see at the clip the cartridges doesn't have profiles so in the next thread i'll add the profiles for the `demo`.
![TIC80](https://user-images.githubusercontent.com/38414584/145521869-35a9683b-3fb2-4d06-84e1-e1fb5f1f948f.png)

https://user-images.githubusercontent.com/38414584/145522130-4980f2e1-3b78-453e-9ab3-788a90ce2087.mp4

**Comment by joshgoebel** [comment]
Nice demo, but probably would be smoother with `--fullscreen`, no?

**Reply by revdeluxe** [reply]
tic80 doesnt want me in fullscreen... but i'll be making a lot of dev vids for this... in YT.
